sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -593,16 +593,27 @@ protected EventProcessor()
 
             if ((diagnosticScope.IsEnabled) && (eventBatch.Any()))
             {
+                bool isBatch = (eventBatch.Count > 1);
```

Comment:
Why declare an interim variable if we're only checking the condition once?

Reply:
We are checking the condition in each iteration of the loop.